storm-core/src/jvm/org/apache/storm/state/KeyValueState.java

```diff
@@ -45,4 +47,9 @@
      * @return the value or defaultValue if no mapping is found
      */
     V get(K key, V defaultValue);
+
+    /**
+     * @return all key value mappings as an unmodifiable map.
+     */
+    Map<K, V> getAll();
```

Comment:
Return an Iterator of the entries in the map than the map itself. That way it will be more manageable even if the state is huge.